src/test/java/com/company/elixr/springbootapplicationwithdocker/FileOperationControllerTest.java

```diff
@@ -0,0 +1,141 @@
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.nio.charset.StandardCharsets;
+
+@RunWith(MockitoJUnitRunner.class)
+@WebMvcTest(FileOperationController.class)
+public class FileOperationControllerTest {
+
+    FileInfo fileInfo = new FileInfo();
+    @Autowired
+    private MockMvc mockMvc;
+    @MockBean
+    private FileOperationService fileOperationService;
+
+    @Before
+    public void init() {
+        MockitoAnnotations.openMocks(this);
+        fileInfo.setFileName("testFile.txt");
+        fileInfo.setUserName("Nathaniel13");
+    }
+
+    @Test
+    public void test_UploadFile() throws Exception {
+
+        MockMultipartFile sampleFile = new MockMultipartFile("file", "testFile.txt",
+                "text/plain",
+                "This is the file content".getBytes(StandardCharsets.UTF_8));
+
+        mockMvc.perform(MockMvcRequestBuilders.multipart("/upload")
+                        .file(sampleFile)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .characterEncoding("UTF-8")
+                        .param("username", "Alex12"))
+                .andExpect(MockMvcResultMatchers.status().isOk());
+    }
+
+    @Test
+    public void test_UploadFile_NoFileProvided() throws Exception {
+
+        mockMvc.perform(MockMvcRequestBuilders.multipart("/upload"))
+                .andExpect(MockMvcResultMatchers.status().isBadRequest());
+    }
+
+    @Test
+    public void test_UploadFile_RequestPart_File_Missing() throws Exception {
+
+        MockMultipartFile sampleFile = new MockMultipartFile("_", "testFile.txt",
+                "text/plain",
+                "This is the file content".getBytes(StandardCharsets.UTF_8));
+
+        mockMvc.perform(MockMvcRequestBuilders.multipart("/upload")
+                        .file(sampleFile)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .characterEncoding("UTF-8")
+                        .param("username", "Alex12"))
+                .andExpect(MockMvcResultMatchers.status().isBadRequest());
+    }
+
+    @Test
+    public void test_UploadFile_RequestPart_UserName_Missing() throws Exception {
+
+        MockMultipartFile sampleFile = new MockMultipartFile("_", "testFile.txt",
+                "document",
+                "This is the file content".getBytes(StandardCharsets.UTF_8));
+
+        mockMvc.perform(MockMvcRequestBuilders.multipart("/upload")
+                        .file(sampleFile)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .characterEncoding("UTF-8")
+                        .param("_", "Alex12"))
+                .andExpect(MockMvcResultMatchers.status().isBadRequest());
+    }
+
+    @Test
+    public void test_UploadFile_RequestParam_UserNameMissing() throws Exception {
```

Comment:
Here userName parameter is not missing, it is empty

```suggestion
    public void test_UploadFile_RequestParam_UserNameEmpty() throws Exception {
```

Reply:
Made the change